report_pre_thesis/src/sections/methodology.tex

```diff
@@ -1,8 +1,246 @@
+However, \citet{cleggRecalibrationMarsScience2017} mentions performing four more types of regression in addition to linear regression, namely parabolic, exponential, geometric and logsquare.
+For each oxide, they used the type of regression and normalization that produced the best results.
+
+% TODO: Move this table to the results section and embed our own results in it as well to show the baseline results compared to the original results.
+% This can be seen in table \ref{tab:regression_types}, which is taken from \citet{cleggRecalibrationMarsScience2017}.
+
+% \begin{table}[h]
+% \centering
+% \begin{tabular*}{\columnwidth}{@{\extracolsep{\fill}}lccc}
+% \toprule
+
+% Element & Norm & Law type    & RMS \\ \midrule
+% Si      & 1    & Logsqaure   & 6.7 \\
+% Ti      & 3    & Geometric   & 0.6 \\
+% Al      & 3    & Geometric   & 4.4 \\
+% Fe      & 1    & Geometric   & 2.2 \\
+% Mg      & 1    & Exponential & 3.0 \\
+% Ca      & 1    & Parabolic   & 1.0 \\
+% Na      & 3    & Parabolic   & 0.6 \\
+% K       & 3    & Geometric   & 0.4 \\
+% \bottomrule
+% \end{tabular*}
+% \caption{The summary of each ICA model characteristics from \citet{cleggRecalibrationMarsScience2017}.}
+% \label{tab:regression_types}
+% \end{table}
+
+% TODO: Update/rewrite this to reflect the new curve-fit implementation.
+% Because \texttt{scikit-learn} does not support these types of regression, we instead opted to transform the data to fit the linear regression model.
+% For example, to perform exponential regression, we take the natural logarithm of the data, perform linear regression, and then take the exponential of the result.
+% This is done for each of the four types of regression mentioned above.
+% The code for this can be seen in listing \ref{lst:data_transformation}.
+
+% \begin{lstlisting}[
+% 	language=Python,
+% 	caption={The code for transforming the data to fit the linear regression model.},
+% 	label={lst:data_transformation}
+% 	]
+% if model_name == "Log-square":
+% 	X_train = np.log(X_train**2)
+% 	X_test = np.log(X_test**2)
+% elif model_name == "Exponential":
+% 	X_train = np.log(X_train)
+% 	X_test = np.log(X_test)
+% elif model_name == "Geometric":
+% 	X_train = np.sqrt(X_train)
+% 	X_test = np.sqrt(X_test)
+% elif model_name == "Parabolic":
+% 	X_train = np.column_stack((X_train, X_train**2))
+% 	X_test = np.column_stack((X_test, X_test**2))
+% \end{lstlisting}
+
+In summary, we have created a replica of the ICA phase of the pipeline with strategic adaptations to accommodate the differences in programming language and the missing or unclear information about the original implementation.
+
+\subsection{MOC}\label{sec:methodology_moc}
+As mentioned in section \ref{sec:moc}, the result of the PLS1-SM and ICA are combined to produce the final MOC model.
+This is done by taking a weighted average of the two models, where the weights are determined based on the fact that one technique is be better suited for certain elements than the other.
```

Comment:
After my comment on this section, I realized that we've already written everything but the weights in 2.1.2. I'd prefer not repeating ourselves. Can we not write something else in this section that bridges the components together nicely?

Reply:
And 2.1.2 is better formulated than here.

Reply:
Agreed. Rephrased to this:

> As described in section \ref{sec:moc_derivation}, the result of the PLS1-SM and ICA are combined to produce the final MOC model, which weights the results of the two techniques in favor of the one that performs best for each oxide.
> The specific weights used for each oxide can be seen in table \ref{tab:weighted_sum_oxide}.
> Note that the weights for Ti, Mg and Ca have been set to 50/50 since \citet{cleggRecalibrationMarsScience2017} does not mention the weights they used for these elements.